src/test/java/com/squareup/kotlinpoet/MemberNameTest.kt

```diff
@@ -0,0 +1,159 @@
+      |package com.example
+      |
+      |import com.squareup.tacos.createTaco
+      |
+      |fun makeTastyTacos() {
+      |    createTaco()
+      |    com.twitter.tacos.createTaco()
+      |}
+      |""".trimMargin())
+  }
+
+  @Test fun memberNamesInsideCompanionsClash() {
+    val squareTacos = ClassName("com.squareup.tacos", "SquareTacos")
+    val twitterTacos = ClassName("com.twitter.tacos", "TwitterTacos")
+    val createSquareTaco = MemberName(squareTacos.nestedClass("Companion"), "createTaco")
+    val createTwitterTaco = MemberName(twitterTacos.nestedClass("Companion"), "createTaco")
+    val file = FileSpec.builder("com.example", "Tacos")
+        .addFunction(FunSpec.builder("makeTastyTacos")
+            .addStatement("%M()", createSquareTaco)
+            .addStatement("%M()", createTwitterTaco)
+            .build())
+        .build()
+    assertThat(file.toString()).isEqualTo("""
+      |package com.example
+      |
+      |import com.squareup.tacos.SquareTacos.Companion.createTaco
+      |import com.twitter.tacos.TwitterTacos
+      |
+      |fun makeTastyTacos() {
+      |    createTaco()
+      |    TwitterTacos.Companion.createTaco()
+      |}
+      |""".trimMargin())
+  }
+
+  @Test fun memberNameAliases() {
+    val createSquareTaco = MemberName("com.squareup.tacos", "createTaco")
+    val createTwitterTaco = MemberName("com.twitter.tacos", "createTaco")
+    val file = FileSpec.builder("com.example", "Tacos")
+        .addAliasedImport(createSquareTaco, "createSquareTaco")
+        .addAliasedImport(createTwitterTaco, "createTwitterTaco")
+        .addFunction(FunSpec.builder("makeTastyTacos")
+            .addStatement("%M()", createSquareTaco)
+            .addStatement("%M()", createTwitterTaco)
+            .build())
+        .build()
+    assertThat(file.toString()).isEqualTo("""
+      |package com.example
+      |
+      |import com.squareup.tacos.createTaco as createSquareTaco
+      |import com.twitter.tacos.createTaco as createTwitterTaco
+      |
+      |fun makeTastyTacos() {
+      |    createSquareTaco()
+      |    createTwitterTaco()
+      |}
+      |""".trimMargin())
+  }
+
+  @Test fun keywordsEscaping() {
+    val `when` = MemberName("org.mockito", "when")
+    val file = FileSpec.builder("com.squareup.tacos", "TacoTest")
+        .addType(TypeSpec.classBuilder("TacoTest")
+            .addFunction(FunSpec.builder("setUp")
+                .addAnnotation(Before::class)
+                .addStatement("%M(tacoService.createTaco()).thenReturn(tastyTaco())", `when`)
+                .build())
+            .build())
+        .build()
+    assertThat(file.toString()).isEqualTo("""
+      |package com.squareup.tacos
+      |
+      |import org.junit.Before
+      |import org.mockito.`when`
+      |
+      |class TacoTest {
+      |    @Before
+      |    fun setUp() {
+      |        `when`(tacoService.createTaco()).thenReturn(tastyTaco())
+      |    }
+      |}
+      |""".trimMargin())
+  }
```

Comment:
can you add a test for a collision between a member name and a class name? It’s gross but possible

Reply:
Good call!